### PR TITLE
Update references from Cloud Logging to Stackdriver Logging

### PR DESCRIPTION
**Pull request by daspecster**
Per: https://github.com/GoogleCloudPlatform/gcloud-common/issues/147

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -20,14 +20,14 @@ This client supports the following Google Cloud Platform services:
 -  `Google Cloud Pub/Sub`_
 -  `Google BigQuery`_
 -  `Google Cloud Resource Manager`_
--  `Google Cloud Logging`_
+-  `Google Stackdriver Logging`_
```
Comment (waprin) [inline README.rst]: I don't *think* it matters but the branding guidelines say long name is "Stackdriver Logging". But the long name for Stackdriver says "Google Stackdriver." So unclear if this should be "Google Stackdriver Logging" or "Stackdriver Logging" @sharbison3 ?

File: README.rst [hunk 2]
```diff
@@ -239,7 +239,7 @@ manage projects using this Client Library.
 
 .. _Resource Manager documentation: https://googlecloudplatform.github.io/gcloud-python/stable/resource-manager-api.html
 
-Google Cloud Logging
+Google Stackdriver Logging
 --------------------
```
Comment (tseaver) [inline README.rst]: Extend the hyphens to match the line length to prevent a Sphinx warning.

File: docs/logging-client.rst [hunk 3]
```diff
@@ -1,4 +1,4 @@
-Logging Client
+Stackdriver Logging Client
 ==============
```
Comment (tseaver) [inline docs/logging-client.rst]: Again, match the line length.

**Comment by tseaver** [comment]
LGTM

**Pull request merged**